test/functional/api/buffer_spec.lua

```diff
@@ -1076,10 +1075,8 @@ describe('api/buf', function()
         -- position the cursor on 't' in 'first'
         api.nvim_win_set_cursor(0, { 1, 19 })
 
-        local cursor = exec_lua([[
-          vim.api.nvim_buf_set_text(0, 0, 15, 2, 24, {'last'})
-          return vim.api.nvim_win_get_cursor(0)
```

Comment:
it is not just about redraw, a test like this specifies that the cursor is valid _immediately_ and not after a separate cycle of normal (or an other input-processing) mode. The alternative, rather than blanked deleting `exec_lua`, would be to use `nvim_call_atomic` or an abstraction built out of it. not sure if that is better than just letting `exec_lua` live though.

Reply:
^ These kinds of comments are useful for future reference, so I un-resolved them. Alternatively (better) is to add them to the commit message.

Reply:
I've added them to the commit message with slight adjustments.